sos/plugins/nvme.py

```diff
@@ -0,0 +1,119 @@
+### This program is free software; you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 2 of the License, or
+## (at your option) any later version.
+
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+
+## You should have received a copy of the GNU General Public License
+## along with this program; if not, write to the Free Software
+## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os, subprocess
+
+class NvmePlugin(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """Collect config and system information about NVMe devices"""
+
+    plugin_name = "nvme"
+    devices = [] 
+
+    def copy_spec(self):
+        """ Loop through all NVMe devices in /sys/block/<nvme-device>/ 
+	and get the files inside queue/, device/, integrity/ and mq/ """
+
+        i = 0
+        while os.path.isdir('/sys/block/nvme%dn1' % i):
+            self.devices.append("nvme%dn1" % i)
+
+            queue_files = subprocess.check_output("ls -1 /sys/block/nvme%dn1/queue/*" % i, shell=True).rstrip()
+            # in python 3.3 or above the prefix b' is not ignored, so bytes.decode() is used to get rid of it
+            queue_files = bytes.decode(queue_files)
```

Comment:
What are you decoding here, and to what? You need to be extremely careful when using text codecs to ensure compatibility for both Python2 and Python3. Generally that means using an explicit codec, disabling error reporting, or a combination of these.